### PR TITLE
remote: Improve testing infrastructure & coverage

### PR DESCRIPTION
**Pull request by jansol**
This adds a pair of wrapper scripts for running (some of) the existing tests with the remote driver using either one or two server daemons (which in turn use the PoCL cpu driver as their native driver).

To kick off slightly more systematic testing, this script is applied to the set of the tests under `tests/runtime` that passed without further modifications.

### CODE REVIEW

File: examples/example1/CMakeLists.txt [hunk 1]
```diff
@@ -94,3 +94,16 @@ set_tests_properties( "examples/example1_poclbin"
     ENVIRONMENT "POCL_KERNEL_CACHE=0"
     LABELS "poclbin;internal;level0"
     DEPENDS "pocl_version_check")
+
+if(ENABLE_REMOTE_CLIENT AND ENABLE_REMOTE_SERVER AND ENABLE_HOST_CPU_DEVICES)
+add_test(NAME "remote/example1"
```
Comment (pjaaskel) [inline examples/example1/CMakeLists.txt:99]: Indent missing?

File: tools/scripts/run_remote_tests [hunk 2]
```diff
@@ -0,0 +1,30 @@
+#!/usr/bin/env bash
+# run_remote_tests - Runs tests against the remote driver, using the PoCL cpu
```
Comment (pjaaskel) [inline tools/scripts/run_remote_tests:2]:
```suggestion
# run_remote_tests - Runs tests against the remote driver, using the PoCL-CPU
```

File: tools/scripts/test_remote_runner_multi.sh [hunk 3]
```diff
@@ -1,25 +1,27 @@
-#!/bin/bash
+#!/usr/bin/env bash
```
Comment (pjaaskel) [inline tools/scripts/test_remote_runner_multi.sh:2]: License text missing.

File: tools/scripts/test_remote_runner_multi.sh [hunk 4]
```diff
@@ -1,25 +1,27 @@
...
 
 BUILD_DIR=$1
+TEST_BINARY=$2
+shift 2
 
-PORT1=$((3000 + $RANDOM % 6000))
-PORT2=$(($PORT1 + 10))
+PORT1=11000
```
Comment (pjaaskel) [inline tools/scripts/test_remote_runner_multi.sh:7]: Is this better than random?

File: tools/scripts/test_remote_runner_single.sh [hunk 5]
```diff
@@ -1,8 +1,10 @@
-#!/bin/bash
+#!/usr/bin/env bash
```
Comment (pjaaskel) [inline tools/scripts/test_remote_runner_single.sh:2]: License

**Review by pjaaskel — APPROVED** [review]
Hm. I thought the ASAN subdevice failure was fixed? Did you not include the fix yet? Perhaps include it in this one?

File: tools/scripts/test_remote_runner_multi.sh (hunk 4 above)
Comment (jansol) [reply tools/scripts/test_remote_runner_multi.sh:7]: We can't guarantee there won't be collisions with random anyway, and this should make errors more deterministic and therefore easier to spot.

**Comment by jansol** [comment]
The ASAN problem *was* fixed, this seems to be different.

**Comment by jansol** [comment]
Couldn't repro the TSAN failure locally but fixed some mildly fishy albeit not obviously problematic accesses in the function the error pointed towards. Also switched the basic tests to RelWithDebInfo instead of Release builds so if it happens again the backtraces should hopefully be a bit more informative.

**Comment by pjaaskel** [comment]
Hm. a subgroups CTS build failure all of a sudden?

**Pull request merged**